src/BlockHorizons/PerWorldPlayer/events/PerWorldPlayerDataSaveEvent.php

```diff
@@ -0,0 +1,23 @@
+<?php
+declare(strict_types=1);
+
+namespace BlockHorizons\PerWorldPlayer\events;
+
+use BlockHorizons\PerWorldPlayer\world\data\PlayerWorldData;
+use BlockHorizons\PerWorldPlayer\world\WorldInstance;
+use pocketmine\event\Cancellable;
+use pocketmine\Player;
+
+class PerWorldPlayerDataSaveEvent extends PerWorldPlayerDataEvent implements Cancellable{
+	/** @var bool */
+	private $quit;
+
+	public function __construct(Player $player, WorldInstance $worldInstance, PlayerWorldData $playerWorldData, bool $quit){
+		parent::__construct($player, $worldInstance, $playerWorldData);
+		$this->quit = $quit;
+	}
+
+	public function hasQuit() : bool{
```

Comment:
Hmm.. would be nicer if this was replaced with `getCause() : int` which returns one of:
```php
public const CAUSE_WORLD_CHANGE = 0;
public const CAUSE_PLAYER_QUIT = 1;
public const CAUSE_CUSTOM = 2;
```
`WorldInstance::save()` method's signature could be changed to:
```php
public function save(Player $player, PlayerWorldData $data, int $cause = PerWorldPlayerDataSaveEvent::CAUSE_CUSTOM, bool $force = false, bool $quit = false) : void;
```

Reply:
Wouldn't $quit be redundant if we can check
 ```php
$ev->getCause() === PerWorldPlayerDataSaveEvent::CAUSE_PLAYER_QUIT
```
?

Reply:
Yup, $cause would take the place of $quit